build-tools/packages/build-cli/src/commands/generate/buildVersion.ts

```diff
@@ -18,11 +18,11 @@ import { BaseCommand } from "../../base";
 export default class GenerateBuildVersionCommand extends BaseCommand<
 	typeof GenerateBuildVersionCommand
 > {
-	static description = `This command is used to compute the version number of Fluid packages. The release version number is based on what's in the lerna.json/package.json. The CI pipeline will supply the build number and branch to determine the prerelease suffix if it is not a tagged build`;
+	static readonly description = `This command is used to compute the version number of Fluid packages. The release version number is based on what's in the lerna.json/package.json. The CI pipeline will supply the build number and branch to determine the prerelease suffix if it is not a tagged build`;
 
-	static examples = ["<%= config.bin %> <%= command.id %>"];
+	static readonly examples = ["<%= config.bin %> <%= command.id %>"];
```

Comment:
This is still mutable since the array provided here is mutable. TypeScript has the feature where you can declare you don't want that interior mutability by adding as const to the end.
```suggestion
	static readonly examples = ["<%= config.bin %> <%= command.id %>"] as const;
```

Reply:
This could be done to all the examples with non-primitive values on the right hand side that infer their type from context (including flags below for example).

Another option is to explicitly state the type instead of inferring it:

```suggestion
	static readonly examples: readonly string[]= ["<%= config.bin %> <%= command.id %>"];
```

Reply:
Good idea. I'll add this tomorrow.

Reply:
`as const` on examples causes a compilation error:

```
Type 'typeof DepsCommand' does not satisfy the constraint 'typeof Command'.
  Types of property 'examples' are incompatible.
    The type 'readonly [{ readonly description: "Bump dependencies on @fluidframework/build-common to the latest release version across all release groups."; readonly command: "<%= config.bin %> <%= command.id %> @fluidframework/build-common -t latest"; }, { ...; }, { ...; }, { ...; }]' is 'readonly' and cannot be assigned to the mutable type 'Example[]'.
```

I think this is coming from oclif typing, not our own base class.

Reply:
as const on the flags property seems to be ok though.